l10n_ch_account_reconcile_isr/models/bank_statement.py

```diff
@@ -0,0 +1,92 @@
+                            select 1 from account_move m
+                            where m.statement_line_id = stl.id
+                            )
+                            AND company_id = %s
+                """
+        params = (self.env.user.company_id.id,)
+        if statements:
+            sql_query += ' AND stl.statement_id IN %s'
+            params += (tuple(statements.ids),)
+        sql_query += ' ORDER BY stl.id'
+        self.env.cr.execute(sql_query, params)
+        st_lines_left = self.env[
+            'account.bank.statement.line'
+        ].browse([line.get('id') for line in self.env.cr.dictfetchall()])
+
+        # try to assign partner to bank_statement_line
+        stl_to_assign_partner = [
+            stl.id for stl in st_lines_left if not stl.partner_id
+        ]
+        refs = list(set([
+            st.name for st in st_lines_left if not st.partner_id
+        ]))
+        acc_cr = st_lines_left[0].journal_id.default_credit_account_id
+        acc_dt = st_lines_left[0].journal_id.default_debit_account_id
+        if st_lines_left and stl_to_assign_partner and refs\
+           and acc_cr and acc_dt:
+
+            sql_query = self._prepare_query()
+            params = (
+                self.env.user.company_id.id,
+                (acc_cr.id, acc_dt.id),
+                tuple(refs)
+            )
+
+            if statements:
+                sql_query += 'AND stl.id IN %s'
+                params += (tuple(stl_to_assign_partner),)
+            self.env.cr.execute(sql_query, params)
+            results = self.env.cr.dictfetchall()
+            st_line = self.env['account.bank.statement.line']
+            for line in results:
+                st_line.browse(
+                    line.get('id')
+                ).write({'partner_id': line.get('partner_id')})
+
+        return {
+            'st_lines_ids': st_lines_left.ids,
+            'notifications': [],
+            'statement_name':
+                len(statements) == 1 and statements[0].name or False,
+            'num_already_reconciled_lines': 0,
+        }
+
+    def _prepare_query(self):
+        """Replace ref on transaction_ref to match aml"""
+        sql_query = """SELECT aml.partner_id, aml.transaction_ref, stl.id
+                FROM account_move_line aml
+                    JOIN account_account acc ON acc.id = aml.account_id
+                    JOIN account_bank_statement_line stl
+                    ON aml.transaction_ref = stl.name
+                WHERE (aml.company_id = %s
+                    AND aml.partner_id IS NOT NULL)
+                    AND (
+                        (aml.statement_id IS NULL AND aml.account_id IN %s)
+                        OR
+                        (acc.internal_type IN ('payable', 'receivable')
+                         AND aml.reconciled = false)
+                        )
+                    AND aml.transaction_ref IN %s
+                    """
```

Comment:
Wouldn't it be better to still search with and return aml.ref if aml.transaction_ref is not set?

Reply:
I updated  to respect ref as well, return value ref or transaction_ref is not important, doesn't used  anywhere